### PR TITLE
[SPARK-33634][SQL][TESTS] Use Analyzer in PlanResolutionSuite

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Instead of using several analyzer rules, this PR uses the actual analyzer to run tests in `PlanResolutionSuite`.

### Why are the changes needed?

Make the test suite to match reality.

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

test-only

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @rdblue @HyukjinKwon

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala [hunk 1]
```diff
@@ -49,6 +51,7 @@ class PlanResolutionSuite extends AnalysisTest {
   private val table: Table = {
     val t = mock(classOf[Table])
     when(t.schema()).thenReturn(new StructType().add("i", "int").add("s", "string"))
+    when(t.partitioning()).thenReturn(Array.empty[Transform])
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:54]: Is this relevant to this Analyzer change?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala [hunk 2]
```diff
@@ -1156,9 +1149,9 @@ class PlanResolutionSuite extends AnalysisTest {
       ("ALTER TABLE testcat.tab ALTER COLUMN i TYPE bigint", false),
       ("ALTER TABLE tab ALTER COLUMN i TYPE bigint", false),
       (s"ALTER TABLE $v2SessionCatalogTable ALTER COLUMN i TYPE bigint", true),
-      ("INSERT INTO TABLE tab VALUES (1)", false),
-      ("INSERT INTO TABLE testcat.tab VALUES (1)", false),
-      (s"INSERT INTO TABLE $v2SessionCatalogTable VALUES (1)", true),
+      ("INSERT INTO TABLE tab VALUES (1, 'a')", false),
+      ("INSERT INTO TABLE testcat.tab VALUES (1, 'a')", false),
+      (s"INSERT INTO TABLE $v2SessionCatalogTable VALUES (1, 'a')", true),
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:1154]: ditto. Is this relevant to this Analyzer change?

**Review by HyukjinKwon — APPROVED** [review]
Looks good. Two same comments from me above too.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:54]: yes, otherwise some tests fail because they insert into the table and need to check the table partition columns.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala (hunk 2 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/command/PlanResolutionSuite.scala:1154]: yes, otherwise some tests fail because the input query has a different number of columns compared to the table schema.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @cloud-fan and @HyukjinKwon .
Merged to master for Apache Spark 3.1
